### PR TITLE
(done) Updated map

### PR DESCRIPTION
**Pull request by nyananu**
Fixed errors with the previous map, and tested it from my end. Please pull and test whenever possible, thanks!

### CODE REVIEW

File: frontend/src/components/profile/Map.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  useEffect(() => {
+    setLocation(currentUser.location)
+  }, [currentUser.location])
+
+  const handleUpdate = async (event) => {
+    event.preventDefault()
+    await dispatch(updateUserAsync({
+      userId,
+      location
+    })).unwrap()
```
Comment (phreid) [inline frontend/src/components/profile/Map.jsx]: does this need to be `await`-ed? can we just dispatch?

**Review by phreid — APPROVED** [review]
works for me. one comment and a suggestion:

if you have time, can you think of a way to give the user some acknowledgement that they've clicked on the map? maybe disable the Update Location button until they click? right now it's kind of hard to know that you're supposed to click a pin, then click update location.

File: frontend/src/components/profile/Map.jsx (hunk 1 above)
Comment (nyananu) [reply frontend/src/components/profile/Map.jsx]: ok, I'll look into these, thanks for reviewing.

**Comment by nyananu** [comment]
Hey @Paul, I removed the await, you were right, we didn't need it. Polished up the edit user modal as well since location will now be handled separately. I tried my best to do the UI interactions possible, but the methods don't seem to be chaining on the onClick function of the marker. I even tried putting in another marker icon, to indicate where the user is currently, but that gave weird behavior (it was only visible when you left and came back to the page, but not visible upon refreshing). 
Feel free to have a look if you want, but for now, I think the map is complete enough to not have errors/warnings, and we can try deploying it into main. I need to do the css and am working with Shirley meanwhile.

File: frontend/src/components/profile/Map.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  useEffect(() => {
+    setUserLocation(currentUser.location)
+  }, [currentUser.location])
+
+  const handleUpdate = (event) => {
+    event.preventDefault()
+    dispatch(updateUserAsync({
+      userId,
+      location: userLocation
+    })).unwrap()
```
Comment (phreid) [inline frontend/src/components/profile/Map.jsx]: small thing here: you shouldn't need the `unwrap` either. otherwise looks good

**Pull request merged**